openverse_catalog/dags/providers/provider_api_scripts/flickr.py

```diff
@@ -59,345 +36,258 @@
-    total_images = 0
-    for image_data in image_list:
-        total_images = _process_image_data(image_data)
-
-    return total_images
-
-
-def _process_image_data(image_data, sub_providers=SUB_PROVIDERS, provider=PROVIDER):
-    logger.debug(f"Processing image data: {image_data}")
-    image_url, height, width = _get_image_url(image_data)
-    license_, license_version = _get_license(image_data.get("license"))
-    creator_url = _build_creator_url(image_data)
-    foreign_id = image_data.get("id")
-    if foreign_id is None:
-        logger.warning("No foreign_id in image_data!")
-    foreign_landing_url = _build_foreign_landing_url(creator_url, foreign_id)
-    owner = image_data.get("owner").strip()
-    source = next((s for s in sub_providers if owner in sub_providers[s]), provider)
-    filesize, filetype = _get_file_properties(image_url)
-    return image_store.add_item(
-        foreign_landing_url=foreign_landing_url,
-        image_url=image_url,
-        license_info=get_license_info(
-            license_=license_, license_version=license_version
-        ),
-        foreign_identifier=foreign_id,
-        width=width,
-        height=height,
-        filesize=filesize,
-        filetype=filetype,
-        creator=image_data.get("ownername"),
-        creator_url=creator_url,
-        title=image_data.get("title"),
-        meta_data=_create_meta_data_dict(image_data),
-        raw_tags=_create_tags_list(image_data),
-        source=source,
-        category=_get_category(image_data),
-    )
-
-
-def _build_creator_url(image_data, photo_url_base=PHOTO_URL_BASE):
-    owner = image_data.get("owner")
-    if owner is not None:
-        creator_url = _url_join(photo_url_base, owner.strip())
-        logger.debug(f"creator_url: {creator_url}")
-    else:
-        logger.warning("No creator_url constructed!")
-        creator_url = None
-
-    return creator_url
 
+    def _url_join(self, *args):
+        return "/".join([s.strip("/") for s in args])
+
+    @staticmethod
+    def _get_largest_image_size(image_data):
+        """
+        Returns the key for the largest image size available.
+        """
+        for size in ["l", "m", "s"]:
+            if f"url_{size}" in image_data:
+                return size
+
+        logger.warning("No image detected!")
+        return None
+
+    @staticmethod
+    def _get_license_info(image_data):
+        license_id = str(image_data.get("license"))
+        if license_id not in LICENSE_INFO:
+            logger.warning(f"Unknown license ID: {license_id}")
+            return None
+
+        license_, license_version = LICENSE_INFO.get(license_id)
+        return get_license_info(license_=license_, license_version=license_version)
+
+    def _get_file_properties(self, image_url):
+        """
+        Get the size of the image in bytes and its filetype.
+        """
+        filesize, filetype = None, None
+        if image_url:
+            filetype = image_url.split(".")[-1]
```

Comment:
If this is just coming from the image URL, should we just remove/comment out this whole function for now since the media store will get the filetype as well?

Reply:
Good point! Yeah, let's just get rid of this entirely.